engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/source/InputMeta.scala

```diff
@@ -0,0 +1,34 @@
+package pl.touk.nussknacker.engine.kafka.source
+
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypedObjectTypingResult, TypingResult}
+
+import scala.reflect.ClassTag
+
+case class InputMeta[K](key: K, topic: String, partition: Int, offset: java.lang.Long, timestamp: java.lang.Long, headers: java.util.Map[String, String])
+
+object InputMeta {
+
+  def withType[K: ClassTag]: typing.TypingResult = {
+    val keyTypingResult = Typed[K]
+    TypedObjectTypingResult(fields(keyTypingResult), objType(keyTypingResult))
+  }
+
+  def withType(keyTypingResult: typing.TypingResult): typing.TypingResult = {
+    TypedObjectTypingResult(fields(keyTypingResult), objType(keyTypingResult))
+  }
+
+  private def fields(keyTypingResult: typing.TypingResult): Map[String, TypingResult] = {
```

Comment:
maybe we could just introduce case class here?

Reply:
We already have `InputMeta` case class. The problem here is that we mix type of key with primitive fields. IMO this way is ok, to resolve this problem. This companion object is in the same file as class so probability of desynchornization is minimal.

Reply:
Fixed (and introduced case class SerializableConsumerRecord)